### PR TITLE
diag

### LINKED ISSUE #15944: diag
Working on (#15024 -> #18061)

### PR DESCRIPTION
**Pull request by Sarvesh-Kesharwani**
Closed #15944

### CODE REVIEW

**Comment by Sarvesh-Kesharwani** [comment]
Need help with the padding value test data generation, everything else is working as expected (tested the function keeping p=st.just(0), with composite strategy), need to generate numeric(int, float) values for it.

File: ivy/functional/frontends/paddle/tensor/creation.py [hunk 1]
```diff
@@ -66,3 +66,13 @@ def arange(start, end=None, step=1, dtype=None, name=None):
...
+
+
+@to_ivy_arrays_and_back
+def diag(x, offset=0, padding_value=0, name=None):
+    d = ivy.diag(x, k=offset)
+    if len(x.shape) == 1:
+        if padding_value:
+            d_mask = ivy.eye(d.shape[0], k=offset)
+            d[d_mask] = padding_value
+    return d
```
Comment (hmahmood24) [inline ivy/functional/frontends/paddle/tensor/creation.py:78]: Try implementing this using `ivy.diagflat` instead. It handles the padding argument for you

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py [hunk 2]
```diff
@@ -302,3 +302,58 @@ def test_paddle_empty(
...
+    return dtype, x, k
+
+
+# diag
+@handle_frontend_test(
+    fn_tree="paddle.diag",
+    # dtype_and_x_k_p=_custom_diag_helper(),
+    dtype_and_x_k=_custom_diag_helper(),
+    # p=helpers.number(min_value=0, max_value=50)
+    p=st.floats(0, 1),
```
Comment (hmahmood24) [inline ivy_tests/test_ivy/test_frontends/test_paddle/test_tensor/test_paddle_creation.py:337]: You can combine multiple strategies using the `st.one_of(..)` strategy. Include the strategies for generating integers `helpers.ints(...)` and `helpers.floats(...)` inside `st.one_of` for pickling randomly from either of these strategies and drawing the test data from that on runtime

**Review by hmahmood24 — CHANGES_REQUESTED** [review]
Hey @Sarvesh-Kesharwani. Reviewed the PR and left comments you can refer to, to improve your implementation. Ask for a re-review once you're done. Thanks! 👍🏼

**Comment by ivy-seed** [comment]
This PR has been labelled as stale because it has been inactive for more than 7 days. If you would like to continue working on this PR, then please add another comment or this PR will be closed in 7 days.

**Comment by ivy-seed** [comment]
This PR has been closed because it has been marked as stale for more than 7 days with no activity.

**Pull request closed without merging**